### PR TITLE
Added support for caching results

### LINKED ISSUE #61: Caching request responses to prevent hitting rate limits
**Description**

GitHub API has rate limits, and sometimes when using starcli it takes up to 10+ secs to see the output, we should cache the responses based on options (like topic/stars/dates etc) somewhere so that when the user uses the same options again within say 1 minute, it can just fetch the cached info instead of sending the request again

**Why is it necessary? (or how would it make it better)**

This improves the execution speed/performance

**how do you think should it be implemented? (If possible)**

Every time search is called we cache the response together with the options used, so when the same options are used again we output the cached data instead

### PR DESCRIPTION
**Pull request by AkashD-Developer**
**Checklist**

- [X] My code is properly formatted using the latest [black](https://github.com/psf/black) 
- [ ] I have added/updated [tests](https://github.com/hedythedev/starcli/tree/main/tests) if needed 
- [X] I have tried running my code manually 
- [X] I have checked for spelling errors 

**Description**
Resolves #61
Added support to cache the responses to a JSON file which will get created when the user first runs the program and also added it to .gitignore so that it will not be tracked by git.

The cache file looks something like this after few requests
cached_results

Those are the keys, which contains the content of the responses.

### CODE REVIEW

**Comment by hedythedev** [comment]
hello, thank you for your pr! I will take a look when I have time, hopefully before the end of next week.

**Comment by AkashD-Developer** [comment]
Thanks, Hedy. Glad to hear back from you. 
Will be waiting for the review.

File: starcli/__main__.py [hunk 1]
```diff
@@ -2,7 +2,10 @@
 
 import click
 import re
+import json
+import os
 
+from datetime import datetime, timedelta
```
Comment (hedythedev) [inline starcli/__main__.py:8]: should be with the other standard library imports

File: starcli/__main__.py [hunk 2]
```diff
@@ -12,6 +15,8 @@
     status_actions,
 )
 
+CACHED_RESULT_FP = os.path.dirname(os.path.dirname(__file__)) + "/.cached_result.json"
```
Comment (hedythedev) [inline starcli/__main__.py:18]: I don't think FP is a good constant name, how about:

```suggestion
CACHED_RESULT_PATH = os.path.dirname(os.path.dirname(__file__)) + "/.cached_result.json"
```

**Review by hedythedev — APPROVED** [review]
LGTM 👍 

There are some minor improvements that could be made, but since I might not be able to review them quickly after your response, I will do them myself. Thanks again!

**Comment by hedythedev** [comment]
@allcontributors please add @AkashD-Developer for code

**Comment by AkashD-Developer** [comment]
Hey Hedy, thank you so much. I would love to contribute more if there’s anything else you require help for.

**Comment by hedythedev** [comment]
No problem, feel free to do the tests ^

**Comment by AkashD-Developer** [comment]
Hey @hedythedev, sure I will take up the new issue. I was actually thinking of some tests for it but was unable to come up with anything. Thanks for the ideas 👍

**Pull request merged**